src/main/java/edu/ucla/library/services/utility/PropsWriter.java

```diff
@@ -0,0 +1,165 @@
+        readWrite(aTemplate, aOutput);
+    }
+
+    /**
+     * Check that parameters are valid.
+     *
+     * @param aArray array of file names
+    */
+    public static void verifyArgs(final String[] aArray) {
+        if (aArray.length != ARGS_COUNT) {
+            System.err.println(
+                "usage: PropsWriter defaults template output");
+            System.exit(Constants.USAGE_ERROR);
+        }
+      // check that default props and template files exist/are readable
+        checkExists(aArray[0]);
+        checkReadable(aArray[0]);
+        checkExists(aArray[1]);
+        checkReadable(aArray[1]);
+    }
+
+    /**
+     * Verify a file exists.
+     *
+     * @param aFileName pathed name of file to test
+    */
+    public static void checkExists(final String aFileName) {
+        if (!Files.exists(FileSystems.getDefault().getPath(aFileName))) {
+            System.err.println("File must exist: " + aFileName);
+            System.exit(Constants.DOESNT_EXIST_ERROR);
+        }
+    }
+
+    /**
+     * Verify a file is readable.
+     *
+     * @param aFileName pathed name of file to test
+    */
+    public static void checkReadable(final String aFileName) {
+        if (!Files.isReadable(FileSystems.getDefault().getPath(aFileName))) {
+            System.err.println("File must be readable: " + aFileName);
+            System.exit(Constants.NOT_READABLE_ERROR);
+        }
+    }
+
+    /**
+     * Load default properties into PROPS.
+     *
+     * @param aFileName pathed name of default props file
+    */
+    public static void loadProps(final String aFileName) {
+        try {
+            PROPS.load(new FileInputStream(new File(aFileName)));
+        } catch (FileNotFoundException e) {
+            System.err.println("Props file must exist: "
+                + e.getMessage());
+            System.exit(Constants.DOESNT_EXIST_ERROR);
+        } catch (IOException details) {
+            System.err.println("I/O error reading props file: "
+                + details.getMessage());
+            System.exit(Constants.IO_ERROR);
+        }
+    }
+
+    /**
+     * Splice default properties in template and write to output file.
+     *
+     * @param aInput pathed name of template file
+     * @param aOutput pathed name of final properties file
+    */
+    public static void readWrite(final String aInput, final String aOutput) {
+        final BufferedReader reader;
+        final BufferedWriter writer;
+        String line = null;
+        try {
+            reader = Files.newBufferedReader(
+                FileSystems.getDefault().getPath(aInput));
+            writer = Files.newBufferedWriter(
+                FileSystems.getDefault().getPath(aOutput));
+            while ((line = reader.readLine()) != null) {
+                if (line.contains(DELIMITER)) {
+                    final String key = line.substring(line.indexOf(DELIMITER) + 1);
+                    writer.write(line.replace(DELIMITER, "").replace(key,
```

Comment:
Here, I think, is where you'd want a System.getenv() check and for it to replace with that value if there is one; else, falling back to the default from the file.